blog/2020-10-08-FacilityLocationProblemQuickstart.adoc

```diff
@@ -0,0 +1,98 @@
+These are the potential *locations* where we can place a facility.
+Each place has a different price that we need to pay before opening a roastery there.
+The price represents the facility *setup cost*.
+
+The challenge is to select a set of places to buy, minimize the total buying cost by preferring cheaper places and, at the same time, minimizing the sum of distances between roasteries and the coffeehouses that they supply.
+
+== The new quickstart
+
+=== Project structure
+
+.Quickstart project structure
+image::facilityLocationProblemProject.png[]
+
+The server side code is in the `src/main/java` directory and is organized in four packages.
+
+bootstrap::
+Generates a randomized (but reproducible) data set and populates the repository with it on application startup.
+
+domain::
+Contains the domain model used by OptaPlanner.
+The `@PlanningSolution`, `@PlanningEntity` and `ConstraintProvider` are in the `domain` package.
+
+persistence::
+Takes care of persisting the facility location problem data set.
+
+rest::
+Contains the REST API used by the client-side code to start and stop solving and to retrieve the best solution.
+
+The client code is in the `src/main/resources/META-INF/resources/` directory.
+It has an `index.html` file that contains the basic HTML structure and loads the `app.js` file that calls server's REST API and makes the web page dynamic.
+
+=== Planning domain model
+
+.Planning domain diagram
+image::facilityLocationProblemClassDiagram.png[]
+
+=== Constraints
+
+The constraints are implemented using the xref:2020-04-07-ConstraintStreams[Constraint Streams] API.
+You can find them in the `org.acme.facilitylocation.domain.FacilityLocationConstraintProvider` class.
+It defines the following constraints:
+
+facilityCapacity::
+Hard constraint that penalizes the demand of consumers assigned to a facility if the total demand exceeds the facility's capacity.
+
+setupCost::
+Soft constraint that penalizes the setup cost of used facility locations.
+
+distanceFromFacility::
+Soft constraint that penalizes the distance of a consumer from its assigned facility.
+
+You can see there are two constraints affecting the soft score level and they might "compete" against each other.
+We need to find a balance between minimizing the distance between consumers and facilities, and minimizing the total setup cost.
+If we put too much weight on the distance constraint we might end up using unnecessarily expensive facilities.
+If we put too much weight on the setup cost constraint the vehicles providing service from facilities to consumers might spend too much time traveling.
+
+//TODO ConstraintConfiguration
+
+
+=== Running the quickstart
+
+Run the quickstart yourself:
+
+. Git clone https://github.com/kiegroup/optaplanner-quickstarts.
+. Descend into the directory `quarkus-facility-location`.
+. Run `mvn quarkus:dev`.
+. Open http://localhost:8080 in your browser and click on the *Solve* button.
```

Comment:
As of today, there is no "stable" branch, so the default is still the "development" branch which uses SNAPSHOTS. I do believe that works with an empty local repo, so this should be fine. But you might want to confirm.
If it's not, let's talk with Radovan: don't add additional instructions here, let's just make sure it's fine out of the box.
If it is, don't mention about SNAPSHOT's etc here: in a few weeks "stable" will be there and this blog will be referenced and read for a few years.

Reply:
Is there any action item for me? I have verified that the steps to run the quickstart work. The procedure appears on multiple places so I have copy-pasted it for consistency. If it needs updating, it should be done for all occurrences in a separate PR.

Reply:
No action item for you - I was just thinking aloud. The current way it's written won't need to be adjusted I believe once we create that "stable" branch and make it the default.